sycl/include/sycl/ext/oneapi/experimental/uniform.hpp

```diff
@@ -0,0 +1,37 @@
+//==------ uniform.hpp - SYCL uniform extension --------*- C++ -*-----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+// Implemenation of the SYCL_EXT_ONEAPI_UNIFORM extension.
+// https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/Uniform/Uniform.asciidoc
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace experimental {
+
+// TODO:
+// An implementation must issue a diagnostic if the
+// sycl::ext::oneapi::experimental::uniform class template is instantiated with
+// any of the following types:
+// sycl::item, sycl::nd_item, sycl::h_item, sycl::group, sycl::sub_group and
+// sycl::nd_range.
+template <class T> class uniform {
```

Comment:
We had feedback from the oneAPI TAB that nothing in this definition makes it obvious that the `uniform` is immutable.  Making a note here that we should see if we can figure out a way to fix this (and update the extension specification) before this gets merged.

Reply:
Thanks for the note. Whenever spec is updated we can update the code.